config/routes.rb

```diff
@@ -22,8 +22,11 @@
     post "/api/saml/auth#{suffix}" => 'saml_post#auth'
     # actual SAML handling POST route
     post "/api/saml/authpost#{suffix}" => 'saml_idp#auth'
+    # The internal auth post which will not be logged as an external request
```

Comment:
This was the only way I could think of to fix the issue I was seeing where an external POST request was being logged 3 times instead of once. If you look at the [code for the log entry](https://github.com/18F/identity-idp/blob/main/app/controllers/saml_idp_controller.rb#L126-L133), it is [checking for the authpost route](https://github.com/18F/identity-idp/blob/main/app/controllers/saml_idp_controller.rb#L72-L80) in the request to determine when to log the external request. I guess we could pass in a query param with something like "external=false", but idk if I love that as a solution either.

Reply:
ah good call - yeah I think this is fine for now - might be worth opening a follow-up ticket to see if there's a way to refactor this a bit more elegantly (e.g. can we track the fact that it's not our first visit another way?), but I don't think this should hold us up from getting this live. Thanks!

Reply:
Good point that we could spend time trying to handle this in another way, given that we know we'll have subsequent requests that mimic the original request. I logged a JIRA to look into that: https://cm-jira.usa.gov/browse/LG-7561 Thanks!